pkg/querier/queryrange/log_result_cache_test.go

```diff
@@ -0,0 +1,452 @@
+package queryrange
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/user"
+
+	"github.com/grafana/loki/pkg/loghttp"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/logqlmodel/stats"
+	"github.com/grafana/loki/pkg/querier/queryrange/queryrangebase"
+	"github.com/grafana/loki/pkg/storage/chunk/cache"
+)
+
+func Test_LogResultCacheSameRange(t *testing.T) {
+	var (
+		ctx = user.InjectOrgID(context.Background(), "foo")
+		lrc = NewLogResultCache(
+			log.NewNopLogger(),
+			fakeLimits{
+				splits: map[string]time.Duration{"foo": time.Minute},
+			},
+			cache.NewMockCache(),
+			nil,
+			nil,
+		)
+	)
+
+	req := &LokiRequest{
+		StartTs: time.Unix(0, time.Minute.Nanoseconds()),
+		EndTs:   time.Unix(0, 2*time.Minute.Nanoseconds()),
+	}
+
+	fake := newFakeResponse([]mockResponse{
+		{
+			RequestResponse: queryrangebase.RequestResponse{
+				Request:  req,
+				Response: emptyResponse(req),
+			},
+		},
+	})
+
+	h := lrc.Wrap(fake)
+
+	resp, err := h.Do(ctx, req)
+	require.NoError(t, err)
+	require.Equal(t, emptyResponse(req), resp)
+	resp, err = h.Do(ctx, req)
+	require.NoError(t, err)
+	require.Equal(t, emptyResponse(req), resp)
+
+	fake.AssertExpectations(t)
```

Comment:
Maybe it would also be useful to assert that there are new entries in the cache. Wdyt?

Reply:
It is kinda already tested by calling it twice and expecting only one call